src/sync-client.js

```diff
@@ -5,1157 +5,1178 @@ var cidProvider = require('./clientIdProvider');
+    },
+
+    checkDatasets: function() {
+      for( var dataset_id in self.datasets ) {
+        if( self.datasets.hasOwnProperty(dataset_id) ) {
+          var dataset = self.datasets[dataset_id];
+          if(dataset && !dataset.syncRunning && (dataset.config.sync_active || dataset.syncForced)) {
+            // Check to see if it is time for the sync loop to run again
+            var lastSyncStart = dataset.syncLoopStart;
+            var lastSyncCmp = dataset.syncLoopEnd;
+            if(dataset.syncForced){
+              dataset.syncPending = true;
+            } else if( lastSyncStart == null ) {
+              self.consoleLog(dataset_id +' - Performing initial sync');
+              // Dataset has never been synced before - do initial sync
               dataset.syncPending = true;
+            } else if (lastSyncCmp != null) {
+              var timeSinceLastSync = new Date().getTime() - lastSyncCmp;
+              var syncFrequency = dataset.config.sync_frequency * 1000;
+              if( timeSinceLastSync > syncFrequency ) {
+                // Time between sync loops has passed - do another sync
+                dataset.syncPending = true;
+              }
             }
-          }
 
-          if( dataset.syncPending ) {
-            // Reset syncForced in case it was what caused the sync cycle to run.
-            dataset.syncForced = false;
+            if( dataset.syncPending ) {
+              // Reset syncForced in case it was what caused the sync cycle to run.
+              dataset.syncForced = false;
 
-            // If the dataset requres syncing, run the sync loop. This may be because the sync interval has passed
-            // or because the sync_frequency has been changed or because a change was made to the dataset and the
-            // immediate_sync flag set to true
-            self.syncLoop(dataset_id);
+              // If the dataset requres syncing, run the sync loop. This may be because the sync interval has passed
+              // or because the sync_frequency has been changed or because a change was made to the dataset and the
+              // immediate_sync flag set to true
+              self.syncLoop(dataset_id);
+            }
           }
         }
       }
-    }
-  },
-
-  /**
-   * Sets cloud handler for sync responsible for making network requests:
-   * For example function(params, success, failure)
-   */
-  setCloudHandler: function(cloudHandler) {
-    if (cloudHandler && typeof cloudHandler === "function") {
-      self.cloudHandler = cloudHandler;
-    } else {
-      self.consoleLog("setCloudHandler called  with wrong parameter");
-    }
-  },
-
-  doCloudCall: function(params, success, failure) {
-    if(self.cloudHandler && typeof self.cloudHandler === "function" ){
-      if (params && params.req) {
-        params.req.__fh = {
-          cuid: cidProvider.getClientId()
-        };
+    },
+
+    /**
+     * Sets clou
```

Comment:
Github diffs are sometimes not very helpful... This line is the only other real change to this file. 

(Also, hooray for abstractions that make this kind of thing super easy to implement!)